### PR TITLE
feat(manager/helmsman): Add OCI support

### PR DESCRIPTION
**Pull request by domolitom**
## Changes

With these changes, the helmsman manager is able to scan OCI registries.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

https://github.com/domolitom/renovate_oci_test

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
Please sign the CLA

File: lib/modules/manager/helmsman/extract.ts [hunk 1]
```diff
@@ -31,6 +35,30 @@ function createDep(
...
+      dep.skipReason = 'no-repository';
+      return dep;
+    }
+
+    dep.datasource = DockerDatasource.id;
+    const ociRegistryUrl = ociRegexResult.groups.registryRef.replace(
+      'oci://',
+      ''
+    );
+    dep.registryUrls = [ociRegistryUrl];
```
Comment (viceice) [inline lib/modules/manager/helmsman/extract.ts:56]: you can simply pass the full image name to packageName without the OCI prefix. makes code simpler

Comment (viceice) [inline lib/modules/manager/helmsman/extract.ts:56]:
```suggestion
```
not needed for docker datasource as it's embedded

File: lib/modules/manager/helmsman/extract.ts [hunk 2]
```diff
@@ -31,6 +35,30 @@ function createDep(
...
+    dep.packageName = ociRegexResult.input.replace('oci://', '');
```
Comment (viceice) [inline lib/modules/manager/helmsman/extract.ts:57]:
```suggestion
    dep.packageName = anApp.chart.replace('oci://', '');
```
we know it's not null.

File: lib/modules/manager/helmsman/extract.ts [hunk 3]
```diff
@@ -31,6 +35,30 @@ function createDep(
...
+  if (ociRegexResult?.groups) {
+    if (!is.nonEmptyString(ociRegexResult.groups.packageName)) {
+      dep.skipReason = 'invalid-name';
+      return dep;
+    }
+
+    if (!is.nonEmptyString(ociRegexResult.groups.registryRef)) {
+      dep.skipReason = 'no-repository';
+      return dep;
+    }
```
Comment (viceice) [inline lib/modules/manager/helmsman/extract.ts:49]: we don't need the regex groups

File: lib/modules/manager/helmsman/extract.ts [hunk 4]
```diff
@@ -31,6 +35,30 @@ function createDep(
   }
   dep.currentValue = anApp.version;
 
+  // in case of OCI repository, we need a PackageDependency with a DockerDatasource and a packageName
+  const ociRegexResult = anApp.chart ? ociChartRegex.exec(anApp.chart) : null;
```
Comment (viceice) [inline lib/modules/manager/helmsman/extract.ts:39]:
```suggestion
  const isOci = anApp?.chart.startsWith('oci://');
```

File: lib/modules/manager/helmsman/extract.ts [hunk 5]
```diff
@@ -31,6 +35,30 @@ function createDep(
...
+  if (ociRegexResult?.groups) {
```
Comment (viceice) [inline lib/modules/manager/helmsman/extract.ts:40]:
```suggestion
  if (isOci) {
```

File: lib/modules/manager/helmsman/extract.ts (hunk 4 above)
Comment (viceice) [reply lib/modules/manager/helmsman/extract.ts:39]: just the starts with

**Review by viceice — DISMISSED** [review]

File: lib/modules/manager/helmsman/extract.ts [hunk 6]
```diff
@@ -31,6 +32,14 @@ function createDep(
   }
   dep.currentValue = anApp.version;
 
+  // in case of OCI repository, we need a PackageDependency with a DockerDatasource and a packageName
+  const isOci = anApp?.chart?.startsWith('oci://');
+  if (isOci) {
+    dep.datasource = DockerDatasource.id;
+    dep.packageName = anApp.chart?.replace('oci://', '');
```
Comment (domolitom) [inline lib/modules/manager/helmsman/extract.ts:39]: The compiler requires anApp.chart to be optional. That was the reason why it failed.

Comment (viceice) [inline lib/modules/manager/helmsman/extract.ts:39]:
```suggestion
    dep.packageName = anApp.chart!.replace('oci://', '');
```
we know it's there here

**Review by viceice — DISMISSED** [review]

**Comment by viceice** [comment]
still coverage issues

File: lib/modules/manager/helmsman/__fixtures__/missingApps.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+namespaces:
```
Comment (viceice) [inline lib/modules/manager/helmsman/__fixtures__/missingApps.yaml:1]: can't we simply use an empty string as test case and inline? so we don't need this fixture?

File: lib/modules/manager/helmsman/__fixtures__/validHelmsfile.yaml [hunk 8]
```diff
@@ -68,11 +74,31 @@ apps:
...
+    enabled: true
+    namespace: test-apps
+    version: 6.4.0
+    chart:
+  podinfo-incorrect-oci:
+    podinfo-incorrect-url:
+    enabled: true
+    namespace: test-apps
+    chart: oci://oci://
+    version: 6.4.0
```
Comment (viceice) [inline lib/modules/manager/helmsman/__fixtures__/validHelmsfile.yaml:104]: I don't think we need these cases for coverage. please revert.

**Review by viceice — APPROVED** [review]

**Pull request merged**